src/box/lua/schema.lua

```diff
@@ -296,16 +298,28 @@ local begin_options = {
         return true
     end,
     txn_isolation = normalize_txn_isolation_level,
+    is_sync = function(is_sync)
+        if type(is_sync) ~= "boolean" then
+            box.error(box.error.ILLEGAL_PARAMS,
+                      "is_sync must be a boolean")
+        end
+        if is_sync == false then
+            box.error(box.error.ILLEGAL_PARAMS, "is_sync can only be true")
+        end
+        return true
+    end,
 }
 
 box.begin = function(options)
     local timeout
     local txn_isolation
-    if options then
+    local is_sync
+    if options ~= nil or options == false then
```

Comment:
Why check `options == false`? Do you mean the code can be called as `box.begin(false)`? Is it a leftover from some previous version of the patch?

Reply:
Ok, I removed that from the check condition. But if the user makes a mistake and writes:
```
tarantool> box.begin(true) box.space.test:insert{5, 'is_sync = nil'} box.commit()
---
- error: Illegal parameters, options should be a table
...

tarantool> box.begin(false) box.space.test:insert{6, 'is_sync = nil'} box.commit()
---
...
```
Would like to be able to warn the user.

Reply:
Then you would also have to check for all other types like strings, numbers, etc. I like the new version better. Where you've started using `check_param_table()`.